lib/model/QubitRepository.php

```diff
@@ -124,14 +124,32 @@ public function save($connection = null)
     return $this;
   }
 
+  public function getRelatedInformationObjectIds()
+  {
+    $sql = "SELECT id FROM ". QubitInformationObject::TABLE_NAME ." WHERE repository_id=:repository_id";
+
+    $results = QubitPdo::fetchAll($sql, array(':repository_id' => $this->id));
+
+    $ioIds = array();
+
+    foreach ($results as $row)
+    {
+      array_push($ioIds, $row->id);
+    }
+
+    return $ioIds;
+  }
+
   public function updateRelatedIos()
   {
     $ioIds = array();
     $context = sfContext::getInstance();
     $env = $context->getConfiguration()->getEnvironment();
 
-    foreach ($this->informationObjects as $io)
+    foreach ($this->getRelatedInformationObjectIds() as $id)
     {
+      $io = QubitInformationObject::getById($id);
```

Comment:
In the case where we are going to update asynchronously (a WebUI call that is going to create a list of IO id's to pass to the async job) it looks like we could omit doing $io = QubitInformationObject::getById($id); to rebuild the array. Can we pass the output of getRelatedInformationObjectIds straight into jobOptions?